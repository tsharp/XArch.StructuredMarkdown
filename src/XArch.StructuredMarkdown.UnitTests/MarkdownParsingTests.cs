using System.Text;

using Snapshooter.Xunit;

using XArch.StructuredMarkdown.Parsing;
using XArch.StructuredMarkdown.Serialization;

namespace XArch.StructuredMarkdown.UnitTests
{
    public class MarkdownParsingTests
    {
        [Fact]
        public void CanParseOnlyFrontmatter()
        {
            string markdown = File.ReadAllText("./test-data/markdown/only_frontmatter.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            document.MatchSnapshot();
        }

        [Fact]
        public void CanParseFullDocumentWithNoFrontMatter()
        {
            string markdown = File.ReadAllText("./test-data/markdown/structured_markdown_no_frontmatter.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            document.MatchSnapshot();
        }

        [Fact]
        public void CanParseAndSerializeFullDocument2()
        {
            string markdown = File.ReadAllText("./test-data/markdown/structured_markdown_with_frontmatter.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            StructuredMarkdownSerializer
                .ToMarkdown(document)
                .MatchSnapshot();
        }

        [Fact]
        public void CanParseAndSerializeFullDocument()
        {
            string markdown = File.ReadAllText("./test-data/markdown/structured_markdown_no_frontmatter.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            StructuredMarkdownSerializer
                .ToMarkdown(document)
                .MatchSnapshot();
        }

        [Fact]
        public void CanParseSection()
        {
            string markdown = File.ReadAllText("./test-data/markdown/section.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            document.MatchSnapshot();
        }

        [Fact]
        public void CanParseParagraph()
        {
            string markdown = File.ReadAllText("./test-data/markdown/paragraph.md");

            StructuredMarkdownDocument document = StructuredMarkdownParser.Parse(markdown);
            document.MatchSnapshot();
        }
    }
}
