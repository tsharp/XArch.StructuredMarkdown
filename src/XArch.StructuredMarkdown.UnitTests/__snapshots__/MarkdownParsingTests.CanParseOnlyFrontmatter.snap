{
  "Root": {
    "Title": null,
    "Parts": [],
    "Metadata": null
  },
  "Metadata": {
    "title": "Structured Markdown Test Document",
    "description": "This document is used for testing the Structured Markdown parser and renderer."
  }
}
